rfc/rfc-48/rfc-48.md

```diff
@@ -0,0 +1,171 @@
+under `Table Path`. This mapping is kept up to date, as operations are performed on the original table. We will leverage
+the same to now maintain mappings to files stored at `Table Storage Path` under different prefixes.
+
+Here are some of the design considerations:
+
+1. Metadata table is a pre-requisite and will be automatically enabled if object storage layout is enabled.
+
+2. The object storage layout cannot be enabled on an existing table that is already bootstrapped with Hive storage layout.
+To switch to object storage layout, the table will need to be re-bootstrapped with the new layout.
+
+3. The Instant metadata (`HoodieCommitMetadata`,`HoodieCleanMetadata` etc.) will always act as the source of file listing
+for metadata table to be populated.
+
+4. `HoodieCommitMetadata` currently stores `file name` instead of complete `file path`. We will have to modify commit
+metadata to store the complete file path instead of just file name, as the files are now distributed across several random
+prefix paths instead of a derivable table/partition path.
+
+5. If there is an error reading from Metadata table, we will not fall back listing from file system.
+
+6. In case of metadata table getting corrupted or lost, we need to have a solution here to reconstruct metadata table
+from the files which distributed using object storage layout. We will likely have to implement a file system listing
+logic, that can get all the partition to files mapping by listing all the prefixes under the `Table Storage Path`.
+Following the folder structure of adding table name/partitions under the prefix will help in getting the listing and
+identifying the table/partition they belong to.
+
+### Query Side Integration
+
+Spark, Hive and [Presto](https://github.com/prestodb/presto/commit/ef1fd25c582631513ccdd097e0a654cda44ec3dc) are
+already integrated to use metadata based listing. Hudi connector for [Trino](https://github.com/prestodb/presto/commit/ef1fd25c582631513ccdd097e0a654cda44ec3dc)
+is currently in process of being contributed, which also integrates with metadata based listing. In general, since these
+query engines are able to use Hudi's metadata table there should ideally be no changes required in
+terms of making them work with Object storage layout. Here are some considerations:
+
+- Spark DataSource and Spark SQL queries have been integrated with metadata based listing via the Hudi's custom implementation
+of Spark's FileIndex interface. However, if Spark DataSource queries are used with globbed paths then the FileIndex path
+does not kick in, and it would rely on Spark's `InMemoryFileIndex` to do the file listing with Hudi's path filter applied. Thus,
+these Spark DataSource queries with globbed paths would not work with object storage layout.
+
+- Query engines should be able to determine that Object storage layout is configured, and rely on metadata to list files. It
```

Comment:
yes we can add a check to fail if storagePath is configured, but metadata table is not